### PR TITLE
GCS_MAVLink: eliminate redundant static pointer to serialmanager

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by WickedShell** [comment]
Ha! I'd made it just far enough to see you manipulate this in the dynamic GCS backend one and was going "why do we need that?" :D

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -107,8 +107,6 @@ GCS_MAVLINK::init(AP_HAL::UARTDriver *port, mavlink_channel_t mav_chan)
 void
 GCS_MAVLINK::setup_uart(const AP_SerialManager& serial_manager, AP_SerialManager::SerialProtocol protocol, uint8_t instance)
```
Comment (WickedShell) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Any reason to not kill the argument and bubble the change the rest of the way up?

Comment (WickedShell) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Added benefit you don't have to pass the protocol into setup_uart realistically, which would let you remove the include for serial manager in GCS.h.

**Comment by peterbarker** [comment]
> Ha! I'd made it just far enough to see you manipulate this in the dynamic GCS backend one and was going "why do we need that?" :D

Check the commit date, 'though.  I've wanted to do this for a while :-)

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: I've added a commit to stop passing the serial manage around.

I've also removed the protocol parameter.

It was removing the include which sold me on it.

Comment (WickedShell) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: > It was removing the include which sold me on it.

I cheated, I figured that would be the compelling argument for you :)

**Comment by peterbarker** [comment]
@WickedShell Also, no more scope-creeping! :-P

**Pull request merged**